File: CMPS12B/Lab2/FileReverse.java
//----------------------------------------------------------------------------------
// FileReverse.java
// Takes two command line arguments, an input and output file.
// Prints each token from the input file backwards on a line by itself to the output
// Alexandra Vareljian
// avarelji
//-----------------------------------------------------------------------------
import java.io.*;
import java.util.Scanner;
import java.lang.*;

class FileReverse{
	public static void main( String args[] ) throws IOException{
		
		//checks for exactly two command line arguments
		if (args.length != 2 ){
			System.out.println("Usage: java -jar FileReverse.jar <input file> <output file>");
			System.exit(1);
		}
		
		//open files
		Scanner in = new Scanner(new File(args[0]));
		PrintWriter out = new PrintWriter(new FileWriter(args[1]));
		
		//reads lines from in, extract tokens, reverse each token, prints tokens
		while( in.hasNextLine() ){
           // trim leading and trailing spaces, then add one trailing space so 
           // split works on blank lines
           String line = in.nextLine().trim() + " "; 

           // split line around white space 
           String[] token = line.split("\\s+");
		
		   //reverse each individual token
		   for(int i=0; i<=token.length-1; i++){
			  token[i] = stringReverse(token[i]);
		   }
		 
		   // print out the now reversed tokens
           for(int i=0; i<=token.length-1; i++){
              out.println(token[i]);
           }
		}
		//close files
		in.close();
		out.close();
	}
      		
	//reverses string by making a character array,
	//switching leftmost and rightmost indexes,
	//moving towards the middle
	public static String stringReverse(String s){
		int l, r = 0;
		char temp;
		//makes the string into an array of chars
		char[] shoobie = s.toCharArray();
		r = shoobie.length-1;
		
		//reverses the chars
		for(l=0; l<r; l++, r--){
			temp = shoobie[l];
			shoobie[l] = shoobie[r];
			shoobie[r] = temp;
		}
		//makes the array of chars back into a string
		s = String.valueOf(shoobie);
		return s;
	}
}
	

File: CMPS12B/Lab3/fileReverse.c
//---------------------------------------------------------------------------------------
// fileReverse.c
// avarelji 1539102
// takes two command line arguments, an in file an out file. prints each string from the
// input file to the output file on a seperate line and reversed
//---------------------------------------------------------------------------------------

#include <stdio.h>
#include<stdlib.h>
#include<string.h>

//swaps first and last characters, moving inward
void stringReverse(char* s){
	int i = 0;
	int z = strlen(s)-1;
	char temp;
	for (i = 0; i<z; i++, z--){
		temp = s[i];
		s[i] = s[z];
		s[z] = temp;
	}
}

int main(int argc, char* argv[]){
	FILE* in;
	FILE* out;
	char word[256]; //a char array to store words from the input file
	
	//check command line for correct number of arguments
	if(argc !=3){
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(EXIT_FAILURE);
	}
	
	//open in file for reading
	in = fopen(argv[1], "r");
	if(in == NULL){
		printf("Unable to read from file %s\n", argv[1]);
		exit(EXIT_FAILURE);
	}
	
	//open output file for writing
	out = fopen(argv[2], "w");
	if(out == NULL){
		printf("Unable to write to file %s\n", argv[2]);
		exit(EXIT_FAILURE);
	}
	
	//read from in, reverse each string, print on seperate lines to out
	while(fscanf(in, " %s", word) != EOF){
		stringReverse(word);
		//printf(word);
		fprintf(out, "%s\n", word);
	}
	
	//close in and out files
	fclose(in);
	fclose(out);
	
	return(EXIT_SUCCESS);
}